fenicsprecice/adapter_core.py

```diff
@@ -22,6 +22,17 @@ class FunctionType(Enum):
     VECTOR = 1  # vector valued function
 
 
+class CouplingMode(Enum):
+    """
+    Defines the type of coupling being used.
+    Options are: Bi-directional coupling, Uni-directional Write Coupling, Uni-directional Read Coupling
+    Used in assertions to check which type of coupling is done
+    """
+    BIDIR = 4
+    UNIDIR_WRITE = 5
+    UNIDIR_READ = 6
```

Comment:
I would suggest to write the full names here. `BIDIRECTIONAL_COUPLING` or `TWO_WAY_COUPLING`. Makes it easier to read. Btw: Why do you start counting at `4`? I think that one `Enum` does not influence the other one.

Additionally: If there is a way to avoid the numbers in the `Enum` this would be great. We don't have to associate numbers with the names, correct? But as far as I know this is a technical detail of python and we have to provide a unique number for each `Enum`. I would be glad, if you proof this statement wrong... Note: There seems to be a way to [solve this using `auto()`](https://docs.python.org/3/library/enum.html#using-automatic-values), but I don't think that this makes things easier to understand and maintain.